### PR TITLE
Log messages for CommClosedError now includes information about remote address

### LINKED ISSUE #5202: Include Peer address in CommClosedError
When a worker in a network dies harshly we often experience a `CommClosedError` raised every other part of the cluster that had a connection to this worker.  To aid debugging I recommend that we include the address of the remote peer that was shut down in the CommClosedError that we raise.  This is often available in `comm._peer_addr`.  

cc @fjetter

### PR DESCRIPTION
**Pull request by fjetter**
Logs like these may pop up on scheduler side. Depending on your logging system it can be quite helpful to know who the other side is.

Closes #5202

### CODE REVIEW

**Comment by fjetter** [comment]
Might close https://github.com/dask/distributed/issues/5202 ?

**Comment by mrocklin** [comment]
> Might close #5202 ?

Maybe there are other places where we raise a CommClosedError?

**Comment by fjetter** [comment]
I went through all logs and raises and adjusted the messages accordingly. Together with https://github.com/dask/distributed/pull/5203 all exceptions should now be much more informative

**Comment by mrocklin** [comment]
Checking in.  What is the status here?

**Comment by mrocklin** [comment]
@crusaderky do you have time to review?  This seems like it would be easy for you.

File: distributed/batched.py [hunk 1]
```diff
@@ -133,7 +133,7 @@ def send(self, *msgs):
         This completes quickly and synchronously
         """
         if self.comm is not None and self.comm.closed():
-            raise CommClosedError()
+            raise CommClosedError(f"Comm {repr(self.comm)} already closed.")
```
Comment (crusaderky) [inline distributed/batched.py:136]:
```suggestion
            raise CommClosedError(f"Comm {self.comm!r} already closed.")
```

File: distributed/core.py [hunk 2]
```diff
@@ -500,9 +500,9 @@ async def handle_comm(self, comm):
                             result = asyncio.ensure_future(result)
                             self._ongoing_coroutines.add(result)
                             result = await result
-                    except (CommClosedError, CancelledError) as e:
+                    except (CommClosedError, CancelledError):
                         if self.status == Status.running:
-                            logger.info("Lost connection to %r: %s", address, e)
+                            logger.info("Lost connection %r", address, exc_info=True)
```
Comment (crusaderky) [inline distributed/core.py:505]:
```suggestion
                            logger.info("Lost connection to %r", address, exc_info=True)
```

**Comment by crusaderky** [comment]
Two cosmetic issues everything else LGTM

**Review by jrbourbeau — APPROVED** [review]
Thanks @fjetter @crusaderky

**Pull request merged**